src/cplscheme/BaseCouplingScheme.hpp

```diff
@@ -97,105 +92,98 @@ class BaseCouplingScheme : public CouplingScheme {
    * hasTimeWindowSize().
    */
-  virtual double getTimeWindowSize() const;
+  virtual double getTimeWindowSize() const override final;
 
   /// returns list of all coupling partners
-  virtual std::vector<std::string> getCouplingPartners() const;
+  virtual std::vector<std::string> getCouplingPartners() const override final;
 
   /**
    * @brief Returns the remaining timestep length within the current time window.
    *
    * If no time window size is prescribed by the coupling scheme, always 0.0 is
    * returned.
    */
-  virtual double getThisTimeWindowRemainder() const;
-
-  /// Returns part of the current time window that has been computed already.
-  virtual double getComputedTimeWindowPart() const
-  {
-    return _computedTimeWindowPart;
-  }
+  virtual double getThisTimeWindowRemainder() const override final;
 
   /**
    * @brief Returns the maximal length of the next timestep to be computed.
    *
    * If no time window size is prescribed by the coupling scheme, always the
    * maximal double accuracy floating point number value is returned.
    */
-  virtual double getNextTimestepMaxLength() const; // TODO mainly used in tests. Is this function actually needed or can we drop it and only use getThisTimeWindowRemainder()?
+  virtual double getNextTimestepMaxLength() const override final; // TODO mainly used in tests. Is this function actually needed or can we drop it and only use getThisTimeWindowRemainder()?
 
   /// Returns true, when the coupled simulation is still ongoing.
-  virtual bool isCouplingOngoing() const;
+  virtual bool isCouplingOngoing() const override final;
 
   /// Returns true, when the accessor can advance to the next time window.
-  virtual bool isTimeWindowComplete() const;
+  virtual bool isTimeWindowComplete() const override final;
 
   /// Returns true, if the given action has to be performed by the accessor.
-  virtual bool isActionRequired(const std::string &actionName) const;
+  virtual bool isActionRequired(const std::string &actionName) const override final;
 
   /// Tells the coupling scheme that the accessor has performed the given action.
-  virtual void markActionFulfilled(const std::string &actionName);
+  virtual void markActionFulfilled(const std::string &actionName) override final;
 
   /// Sets an action required to be performed by the accessor.
-  virtual void requireAction(const std::string &actionName);
+  virtual void requireAction(const std::string &actionName) override final;
 
   /// Finalizes the coupling scheme.
-  virtual void finalize();
+  virtual void finalize() override final;
 
   /**
-* @brief Initializes the coupling scheme.
-*
-* @param[in] startTime TODO
-* @param[in] startTimeWindow TODO
-*/
-  virtual void initialize(double startTime, int startTimeWindow) = 0;
+   * @brief Initializes the coupling scheme.
+   *
+   * @param[in] startTime starting time of coupling scheme
+   * @param[in] startTimeWindow ID of time window, from which coupling scheme starts
```

Comment:
What do you mean by "ID"? These arguments are a relic of past times when preCICE had a restart mode. I kept them there to eventually re-enable the restart.

Reply:
As far as I understand the variable it stores the coupling windows' counter/ID. Is this true? I'm open for suggestions.

Reply:
I think we mean the same thing, it's the number of the coupling window, but "ID" is just an odd name for this, isn't it?

Reply:
I agree that it is an odd name ;) The only alternative I have in mind is "counter", but I do not really like it, as well.

Reply:
Well, we don't want to change the name, right? `startTimeWindow` is perfectly fine. We just want to change the documentation. 
What about "Starting counter of time window"?

Reply:
I like this. Changed it in f094f1f6905016c9cb310660c097954b1653e168.